test/unit/authInfoTest.ts

```diff
@@ -2014,6 +2014,19 @@ describe('AuthInfo', () => {
       expect(options.loginUrl).to.equal('https://test.my.salesforce.com');
     });
 
+    it('should parse the correct url when passed in as json', () => {
```

Comment:
do we have a test for the case it's a just a string?

Reply:
Yes, every other test for this function uses just a string.
For example, https://github.com/forcedotcom/sfdx-core/pull/387/files#diff-72e39279bdd6c1fae17308917208a797a82022b0150c99572b109abcffe4449fR1990